tests: Add some workloads

fabiand opened the pull request.

fabiand committed:
tests: Add some workloads

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
tests: Include workloads

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
tests: Verify that there is load

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
tests: Add fixme

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand commented:
@tiraboschi please take a look

fabiand committed:
tests: Oops

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
to: Add deployment of operators

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

tiraboschi commented on tests/01-vms-cpu-load.yaml:
```diff
@@ -0,0 +1,55 @@
+          labels:
+            kubevirt.io/vmpool: cpu-load
+        spec:
+          domain:
```
To generate a relevant amount of CPU pressure on a node, we probably have to require more CPU cores here (`xz -T 0` is already multi-thread): e.g. even if 5 fully loaded VMs with 1 core each land of the same physical node with 8 physical cores we are still not generating enough CPU load to stress it.

fabiand replied:
This got fixed by manuall setting guest cores and cpu requests.

fabiand committed:
tests: Increase the load slowly

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
to: Add operator deployment and installation

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
deploy: Retry if webhooks are not ready yet

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
test: Cleanup deploy

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
test: Load generation is working

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
tests: Start working on rebalancing test

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
tests: Tuning

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
test: Finanlly works

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
test: More stabilization

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
Further tuning.

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
desched-taint: Do node tainting based on descheduler output

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
desched-taint: Remove taints on first master visit

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
tests: Fine tuning

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
deched-taint: Better cycle handling

Avoids races where master comes later

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

tiraboschi commented on e2e-test.sh:
```diff
@@ -2,44 +2,109 @@
-x "bash to.sh wait_for_mcp"
+c "Configure decsheduler for automatic mode and faster rebalancing"
+x "oc patch --type=json -p '[{\"op\": \"replace\", \"path\": \"/spec/mode\", \"value\": \"Automatic\"}]' -n openshift-kube-descheduler-operator KubeDescheduler cluster"
+x "oc patch --type=json -p '[{\"op\": \"replace\", \"path\": \"/spec/deschedulingIntervalSeconds\", \"value\": 12}]' -n openshift-kube-descheduler-operator KubeDescheduler cluster"
```
Why do we have to reduce it to 12? Is it just for speed up the test process?

fabiand replied:
Fixed in follow up commits

tiraboschi commented on e2e-test.sh:
```diff
@@ -2,44 +2,109 @@
+if $WITH_DEPLOY; then x "bash to.sh deploy"; fi
+
+n
+c "Taint node for in-balance"
```
I think we need also a cleanup function (removing our taint from all the nodes and the two virtualmachinepools) to be executed before starting the test and on exit with trap to try ensuring we always have a clean and reproducible environment without leftovers.

fabiand replied:
I think you provided it by now ;)

fabiand committed:
tainter: Move tainter to deployment

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
tainter: Add to deployment

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
test: Increase load slower

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
Add URL to OCP Console for watching metrics

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
test: Hard code CPU model in order to keep migration running

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

tiraboschi commented on e2e-test.sh:
```diff
@@ -2,44 +2,109 @@
+
+c "Let the descheduler run for a bit in order to rebalance the cluster"
+c "Use the following URL in order to monitor key metrics"
+echo $(oc get console cluster -o=jsonpath='{@.status.consoleURL}')'/monitoring/query-browser?query0=sum+by+(instance)+(rate(node_pressure_cpu_waiting_seconds_total{instance%3D~".*worker.*"}[1m]))&query1=count+by+(node)+(kubevirt_vmi_info{node%3D~".%2B"%2Cname%3D~"cpu.*"})+>+0&query2=stddev(sum+by+(instance)+(rate(node_pressure_cpu_waiting_seconds_total{instance%3D~".*worker.*"}[1m])))'
```
```suggestion
echo $(oc get console cluster -o=jsonpath='{@.status.consoleURL}')'/monitoring/query-browser?query0=rate(node_pressure_cpu_waiting_seconds_total{instance%3D~".*worker.*"}[1m])&query1=count+by+(node)+(kubevirt_vmi_info{node%3D~".*worker.*"%2Cname%3D~"cpu.*"})+>+0&query2=stddev(sum+by+(instance)+(rate(node_pressure_cpu_waiting_seconds_total{instance%3D~".*worker.*"}[1m])))'
```

fabiand replied:
Fixed with your query. Thanks

fabiand committed:
tatiner: tune

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
tests: Tuning

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
tests: We can not make assumptions about how many noes will be used, because non-VM pressure can lead VMs to be pushed away

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
iREADME small updaze

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
ito/tests: Rework logic to not rely on worker in instance and node names

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
tainter: Fix wrt predictive/auto mde

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>

fabiand committed:
tainter: Fix and disable

It turns out that the tainter does not yield significant performanece on larger clusters

Signed-off-by: Fabian Deutsch <fabiand@fedoraproject.org>
